File: python/factorial.py
#!/usr/bin/python

import sys

def factorial(n):
    retValue = 1
    for i in range(2, n + 1):
        retValue *= i
    return(retValue)

if '__main__' == __name__:
    assert 2 == len(sys.argv)
    print(factorial(int(sys.argv[1])))

File: ruby/factorial.rb
#!/usr/bin/ruby

def factorial(n)
  retValue = 1
  2.upto(n) do |i|
    retValue *= i
  end
  puts retValue
end

if 1 != ARGV.size()
  puts 'please provide exactly one argument'
else
  factorial(ARGV.pop().to_i)
end
